testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/naturalid/HibernateNativeAPINaturalIdTestCase.java

```diff
@@ -117,9 +125,24 @@ public static void beforeClass() throws NamingException {
                 + " <module name=\"com.h2database.h2\" />" + " <module name=\"org.slf4j\"/>" + " </dependencies>"
                 + " </deployment>" + "</jboss-deployment-structure>"), "jboss-deployment-structure.xml");
 
+        final String baseModulesPath = System.getProperty("jboss.dist") + turnToPath("/modules/system/layers/base/");
+        final String basedir = System.getProperty("basedir");
+        ear.addAsManifestResource(
+                createPermissionsXmlAsset(new ReflectPermission("suppressAccessChecks"), new RuntimePermission(
+                        "createClassLoader"), new RuntimePermission("setContextClassLoader"), new RuntimePermission(
+                        "getClassLoader"), new RuntimePermission("accessDeclaredMembers"), new RuntimePermission(
+                        "getProtectionDomain"), new PropertyPermission("user.dir", "read"), new FilePermission(basedir
```

Comment:
This does not look like the correct fix. Why does this test need these permissions?

In general if a test fails with a security manager exception unless the test code is directly doing something that needs permission it should be fixed in our code, not just covered up with permissions.xml.

Reply:
Agree, the Hibernate part is fixed, component upgrade is necessary. See https://hibernate.atlassian.net/browse/HHH-11615

Reply:
OK, so I'll do the following: I will use (locally) the Hibernate 5.2.12 (that includes HHH-11615 fix), and see how this reduce the number of permissions needed.